src/model/model.nit

```diff
@@ -784,6 +784,18 @@ class MClassDef
 	# All property introductions and redefinitions (not inheritance) in `self` by its associated property.
 	var mpropdefs_by_property = new HashMap[MProperty, MPropDef]
 
+	# Return the direct supertypes of the `mclassdef`
```

Comment:
improve the doc. I don't understand what it really does. give an example maybe

Reply:
Example:

```nit

	module 1

		class A

		
		class B
			super A



	module 2

		redef class A


		class C
			super B

mclassdef_C.get_direct_superparents == [B]
```

In our context when we evaluate the C class to define its default constructor, we need to get the constructors of the direct parents to retrieve their initializers. 
One way to get the direct parents is to use `in_hierarchy.direct_greater` of MClass the problem is that every time we use this method we have to flatten the class hierarchy which has a significant cost (about + 30% on the compilation time). 
The second method is to use `in_hierarchy.direct_greater` of mclassdef, the main problem is that if we deal with the direct parents mclassdef, we will get the redef of A and B class. To eliminate these cases we remove from the list of direct parent all superparent of potential direct parents returned by `in_hierarchy.direct_greater`. In other word `in_hierarchy.direct_greater` of C return [A,B]. In this return you can remove A because B is also an A. We can say that the direct parent of C is B.